src/eduid_common/api/helpers.py

```diff
@@ -126,32 +139,37 @@ def verify_nin_for_user(user, proofing_state, proofing_log_entry):
     if nin_element and nin_element.is_verified:
         current_app.logger.info('NIN is already verified for user {}'.format(proofing_user))
         current_app.logger.debug('NIN: {}'.format(proofing_state.nin.number))
-        return
+        return True
 
     # Update users nin element
     if proofing_user.nins.primary is None:
         # No primary NIN found, make the only verified NIN primary
         nin_element.is_primary = True
     nin_element.is_verified = True
-    nin_element.verified_ts = True
+    # Ensure matching timestamp in verification log entry, and NIN element on user
+    nin_element.verified_ts = proofing_log_entry.created_ts
     nin_element.verified_by = proofing_state.nin.created_by
 
     # Update users name
     proofing_user = set_user_names_from_offical_address(proofing_user, proofing_log_entry)
 
     # If user was updated successfully continue with logging the proof and saving the user to central db
     # Send proofing data to the proofing log
-    if current_app.proofing_log.save(proofing_log_entry):
-        current_app.logger.info('Recorded verification for {} in the proofing log'.format(proofing_user))
-        # User from central db is as up to date as it can be no need to check for modified time
-        proofing_user.modified_ts = True
-        # Save user to private db
-        current_app.private_userdb.save(proofing_user, check_sync=False)
+    if not current_app.proofing_log.save(proofing_log_entry):
+        return False
 
-        # Ask am to sync user to central db
-        current_app.logger.info('Request sync for user {!s}'.format(user))
-        result = current_app.am_relay.request_user_sync(proofing_user)
-        current_app.logger.info('Sync result for user {!s}: {!s}'.format(proofing_user, result))
+    current_app.logger.info('Recorded verification for {} in the proofing log'.format(proofing_user))
+    # User from central db is as up to date as it can be no need to check for modified time
```

Comment:
"central db" -> "private db"?

Reply:
I changed it to in fact check sync instead.